src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentPresentation/TextDocumentUriPresentationEndpoint.cs

```diff
@@ -131,16 +131,25 @@ protected override IRazorPresentationParams CreateRazorRequestParameters(UriPres
         {
             _logger.LogInformation("Trying to find document info for dropped uri {uri}.", uri);
 
-            var documentContext = await _documentContextFactory.TryCreateAsync(uri, cancellationToken).ConfigureAwait(false);
```

Comment:
Interesting, can you expand on why the old approach didn't work?

Reply:
I can! https://github.com/dotnet/razor-tooling/issues/6960

Should come up in triage if anything isn't clear